Update Dependabot to ignore patch updates

jbangelo opened the pull request:
We tend to get a bunch of patch updates to dependencies , which can create a lot of pointless noise in the commit history. Since we have security scans enabled, it should be fine to skip over patch updates until we see encounter a bug that requires an update.

This also adds a reviewer to the Dependabot PRs, so that we can get them reviewed and merged in a timely fashion.

jbangelo committed:
Update Dependabot to ignore patch updates

jbangelo committed:
Address Copilot suggestion

john-michaelburke reviewed (approved):
you could probably throw a group on there like the quality group or something unless you'd prefer to be blasted on your own. Seems like we could benefit from like a generic rust reviewer group which you could use for things like this

pcrumley reviewed (changes requested):
I hear where you are coming from but this is a little heavy handed in your approach.

at the very least dependabot knows security updates vs patch and we should allow those. I also think it is nice to group updates to avoid spam. I found this setting works well for my rust repos:
```yaml
---
version: 2
updates:
  - package-ecosystem: "cargo"
    directory: "/"
    schedule:
      interval: "daily"
    groups:
      minor-and-patch:
        applies-to: version-updates
        update-types:
          - patch
          - minor
      security-updates:
        applies-to: security-updates
        update-types:
          - patch
          - minor
  - package-ecosystem: github-actions
    directory: "/"
    schedule:
      interval: weekly
```

pcrumley commented on .github/dependabot.yml:
```diff
@@ -6,13 +6,20 @@ updates:
     - 0.4.36
+  # Ignore patch updates for all
+  - dependency-name: "*"
+    update-types: ["version-update:semver-patch"]
```
actually this already only affects version-updates. sorry for being trigger happy with changes requested. I think it makes sense to group instead of disabling if you are worried about git log spam but this is fine as is if you are happy with it.

jbangelo replied:
Maybe doing less often checks in addition to grouping them together would be the right path forward?

pcrumley replied:
@jbangelo that's what I would recommend, but i don't feel strongly about it.

pcrumley commented:
fyi there isn't really a good reason to keep vergen pinned at version three. I updated it to the latest version in line with other rust tools we have: https://github.com/swift-nav/ntripping/pull/272

pcrumley committed:
Merge branch 'master' into jbangelo/skip-patch-versions

jbangelo committed:
Switch out the default reviewer

jbangelo commented:
> you could probably throw a group on there like the quality group or something unless you'd prefer to be blasted on your own. Seems like we could benefit from like a generic rust reviewer group which you could use for things like this

Changed out me for the @swift-nav/open-source group, just had to make sure we had added that team to this repo.
